SimPEG/PF/Gravity.py

```diff
@@ -178,301 +133,201 @@ def Intrgl_Fwr_Op(self, flag):
+                            dy[:, bb] * np.log(dz[:, cc] + r + eps) +
+                            dz[:, cc] * np.log(dy[:, bb] + r + eps) -
+                            dx[:, aa] * np.arctan(dy[:, bb] * dz[:, cc] /
+                                                  (dx[:, aa] * r + eps)))
 
-        X, Y = np.meshgrid(x, y)
+                    elif self.rxType == 'y':
+                        row -= NewtG * (-1) ** aa * (-1) ** bb * (-1) ** cc * (
+                            dx[:, aa] * np.log(dz[:, cc] + r + eps) +
+                            dz[:, cc] * np.log(dx[:, aa] + r + eps) -
+                            dy[:, bb] * np.arctan(dx[:, aa] * dz[:, cc] /
+                                                  (dy[:, bb] * r + eps)))
 
-        # Interpolate
-        d_grid = griddata(rxLoc[:, 0:2], d, (X, Y), method='linear')
-        plt.imshow(d_grid, extent=[x.min(), x.max(), y.min(), y.max()],
-                   origin='lower', vmin=vmin, vmax=vmax, cmap="plasma")
-        plt.colorbar(fraction=0.02)
+                    else:
+                        row -= NewtG * (-1) ** aa * (-1) ** bb * (-1) ** cc * (
+                            dx[:, aa] * np.log(dy[:, bb] + r + eps) +
+                            dy[:, bb] * np.log(dx[:, aa] + r + eps) -
+                            dz[:, cc] * np.arctan(dx[:, aa] * dy[:, bb] /
+                                                  (dz[:, cc] * r + eps)))
 
-        if levels is None:
-            plt.contour(X, Y, d_grid, 10, vmin=vmin, vmax=vmax, cmap="plasma")
+        if self.forwardOnly:
+            return np.dot(row, self.model)
         else:
-            plt.contour(X, Y, d_grid, levels=levels, colors='r',
-                        vmin=vmin, vmax=vmax, cmap="plasma")
-
-    plt.title(varstr)
-    plt.gca().set_aspect('equal', adjustable='box')
-
-    return fig
-
-
-def readUBCgravObs(obs_file):
-
-    """
-    Read UBC grav file format
-
-    INPUT:
-    :param fileName, path to the UBC obs grav file
+            return row
 
-    OUTPUT:
-    :param survey
-
-    """
-
-    fid = open(obs_file, 'r')
-
-    # First line has the number of rows
-    line = fid.readline()
-    ndat = np.array(line.split(), dtype=int)
-
-    # Pre-allocate space for obsx, obsy, obsz, data, uncert
-    line = fid.readline()
-    temp = np.array(line.split(), dtype=float)
-
-    d = np.zeros(ndat, dtype=float)
-    wd = np.zeros(ndat, dtype=float)
-    locXYZ = np.zeros((ndat, 3), dtype=float)
+    def progress(self, ind, total):
+        """
+        progress(ind,prog,final)
 
-    for ii in range(ndat):
+        Function measuring the progress of a process and print to screen the %.
+        Useful to estimate the remaining runtime of a large problem.
 
-        temp = np.array(line.split(), dtype=float)
-        locXYZ[ii, :] = temp[:3]
-        d[ii] = temp[3]
-        wd[ii] = temp[4]
-        line = fid.readline()
+        Created on Dec, 20th 2015
```

Comment:
here, we don't need to track date and author info, this is kept track of by github (for examples it is helpful, but in the main codebase, we don't really need to)